crypto/ec/ecdsa_ossl.c

```diff
@@ -99,14 +105,17 @@ static int ecdsa_sign_setup(EC_KEY *eckey, BN_CTX *ctx_in,
 
         /*
          * We do not want timing information to leak the length of k, so we
-         * compute G*k using an equivalent scalar of fixed bit-length.
+         * compute G^k using an equivalent scalar of fixed bit-length.
+         *
+         * We unconditionally perform both of these additions to prevent a
+         * small timing information leakage.  We then choose the sum that is
+         * one bit longer than the order.  This guarantees the code
+         * path used in the constant time implementations elsewhere.
          */
-
-        if (!BN_add(k, k, order))
+        if (!BN_add(r, k, order)
+            || !BN_add(X, r, order)
+            || !BN_copy(k, BN_num_bits(r) > BN_num_bits(order) ? r : X))
```

Comment:
Maybe same TODO commentary note even here? Well, it's not point to inject commentary specifically prior BN_copy, perhaps add few words above...

Reply:
I've added TOTO comments in both places.